### PR TITLE
Meta: Improve build speed and avoid redundant type issues

### PR DESCRIPTION
**Pull request by kovsu**

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -38,6 +38,7 @@ const rollup = {
 			return chunkInfo.name.replace('build/__snapshots__/', '') + '.js';
 		},
 	},
+	context: 'window',
```
Comment (kovsu) [inline rollup.config.js:41]: Avoid this warning. https://rollupjs.org/troubleshooting/#error-this-is-undefined
image

File: rollup.config.js [hunk 2]
```diff
@@ -57,11 +58,8 @@ const rollup = {
 				{find: 'react', replacement: 'dom-chef'},
 			],
 		}),
-		typescript({
-			compilerOptions: {
-				noEmitOnError: !process.env.ROLLUP_WATCH,
-				module: 'Node16',
-			},
+		sucrase({
+			transforms: ['typescript', 'jsx'],
```
Comment (kovsu) [inline rollup.config.js:62]:
```js
exclude: ['node_modules/**'],
transforms: ['typescript', 'jsx'],
```

I prefer the config like this, but the `octicons-plain-react` library has exported `tsx` file, so I can't exclude `node_modules`. If you update the`octicons-plain-react`, I think maybe we can use `rollup-plugin-esbuild` to replace `@rollup/plugin-sucrase`.

**Review by fregante — APPROVED** [review]
Wow, this speeds up the build considerably.

File: rollup.config.js (hunk 1 above)
Comment (fregante) [reply rollup.config.js:41]: We should add a link to https://github.com/sindresorhus/memoize/issues/102 so we will drop this after the change is done. 
```suggestion
   // TODO: blah blah
	context: 'window',
```

File: rollup.config.js (hunk 2 above)
Comment (fregante) [reply rollup.config.js:62]: Given how fast it seems to be, I think it’s good as is. As soon as Vitest switches to Rolldown or [esbuild supports `preserveModules`](https://github.com/evanw/esbuild/issues/708) we might change bundler again anyway.

**Comment by kovsu** [comment]
Oh, I forget to update that. 😂 Sooooorry

**Comment by fregante** [comment]
Tested, the extension works

**Pull request merged**